docs/design/kubectl-oadp.md

```diff
@@ -0,0 +1,74 @@
+# Kubectl-oadp plugin design
+
+## Abstract
+The purpose of this plugin is to allow the customer to delete backups, along with creating restores in OADP without needing to alias velero to do so.
+
+## Background
+The current OADP cli is suboptimal as oc backup delete $foo deletes the k8 object instead of the backup but velero backup delete $foo deletes the backup, along with the backup files in storage. Currently, customers would need to alias velero in order to delete their backups, which is not ideal. The purpose of kubectl-oadp would be to make the cli experience better and easier to use along with enabling users to be able to get the logs of the backups.
+
+## Goals
+- Customers can create backups and restores
+- A non-cluster admin can create Non-Admin-Backups (NAB)
```

Comment:
Would the non-admin backup/restore API calls be different? Since the CRDs are different, I imagine those would need different APIs here as well.